python-primitiv/primitiv/_graph.pyx

```diff
@@ -148,7 +148,9 @@ cdef class _Node:
     def __pow__(left, right, mod):
         if mod is not None:
             return NotImplemented
-        if isinstance(right, (int, float)):
+        if isinstance(right, int) and right >= 0:
```

Comment:
The value assertion about `right` should be fixed:
* It should accept negative integers.
* It should reject overflow-ed values because Python3 does not distinguish 32-bits integers and longer integers.